### PR TITLE
Added support for AtomicInteger, AtomicBoolean, AtomicLong and Atomic…

### PR DESCRIPTION
**Pull request by inder123**
…IntegerArray.

### CODE REVIEW

File: gson/src/test/java/com/google/gson/functional/JavaUtilConcurrentLocksTest.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicIntegerArray;
+import java.util.concurrent.atomic.AtomicLong;
+
+import com.google.gson.Gson;
+
+import junit.framework.TestCase;
+
+/**
+ * Functional test for Json serialization and deserialization for java.util.concurrent.locks classes
```
Comment (swankjesse) [inline gson/src/test/java/com/google/gson/functional/JavaUtilConcurrentLocksTest.java]: don't you mean `java.util.concurrent.atomic` ?

**Comment by swankjesse** [comment]
LGTM

File: gson/src/test/java/com/google/gson/functional/JavaUtilConcurrentLocksTest.java (hunk 1 above)
Comment (inder123) [inline gson/src/test/java/com/google/gson/functional/JavaUtilConcurrentLocksTest.java]: fixed

**Comment by inder123** [comment]
Self-merging since @swankjesse  LGTM'd it.

**Pull request merged**